src/main/java/com/massivecraft/factions/data/MemoryBoard.java

```diff
@@ -272,81 +272,84 @@ public int getFactionCoordCountInWorld(Faction faction, String worldName) {
-
-        // Get the compass
-        ArrayList<String> asciiCompass = AsciiCompass.getAsciiCompass(inDegrees, ChatColor.RED, FactionsPlugin.getInstance().txt().parse("<a>"));
-
-        int halfWidth = FactionsPlugin.getInstance().conf().map().getWidth() / 2;
-        // Use player's value for height
-        int halfHeight = fplayer.getMapHeight() / 2;
-        FLocation topLeft = flocation.getRelative(-halfWidth, -halfHeight);
-        int width = halfWidth * 2 + 1;
-        int height = halfHeight * 2 + 1;
-
-        if (FactionsPlugin.getInstance().conf().map().isShowFactionKey()) {
-            height--;
-        }
+    public ArrayList<FancyMessage> getMap(FPlayer fPlayer, FLocation fLocation, double inDegrees) {
+        final MainConfig conf = FactionsPlugin.getInstance().conf();
+        final ArrayList<FancyMessage> out = new ArrayList<>();
 
-        Map<String, Character> fList = new HashMap<>();
-        int chrIdx = 0;
+        final Faction senderFaction = fPlayer.getFaction();
+        final Faction senderFactionAtLocation = getFactionAt(fLocation);
 
-        // For each row
-        for (int dz = 0; dz < height; dz++) {
-            // Draw and add that row
-            FancyMessage row = new FancyMessage("");
+        out.add(new FancyMessage(FactionsPlugin.getInstance()
+            .txt().titleize("(" + fLocation.getCoordString() + ") " + senderFactionAtLocation.getTag(fPlayer))));
 
-            if (dz < 3) {
-                row.then(asciiCompass.get(dz));
-            }
-            for (int dx = (dz < 3 ? 6 : 3); dx < width; dx++) {
-                if (dx == halfWidth && dz == halfHeight) {
-                    row.then("+").color(ChatColor.AQUA);
-                    if (false) {
-                        row.tooltip(TL.CLAIM_YOUAREHERE.toString());
+        final ArrayList<String> compass =
+            AsciiCompass.getAsciiCompass(inDegrees, ChatColor.RED, FactionsPlugin.getInstance().txt().parse("<a>"));
+
+        final int halfMapWidth = conf.map().getWidth() / 2;
+        final int halfMapHeight = fPlayer.getMapHeight() / 2;
+        final int mapWidth = halfMapWidth * 2 + 1;
+        final int mapHeight = conf.map().isShowFactionKey() ? halfMapHeight * 2 : halfMapHeight * 2 + 1;
+
+        final FLocation topLeft = fLocation.getRelative(-halfMapWidth, -halfMapHeight);
+        final Map<String, Character> factionsList = new HashMap<>();
+
+        int charIndex = 0;
+
+        // ... for each row
+        for (int dz = 0; dz < mapHeight; dz++) {
+            final FancyMessage row = new FancyMessage("");
+
+            // ... add ascii compass
+            if (dz < 3) row.then(compass.get(dz));
+
+            for (int dx = (dz < 3 ? 6 : 3); dx < mapWidth; dx++) {
+                // ... current chunk fplayer is standing in
+                if (dx == halfMapWidth && dz == halfMapHeight) {
+                    row.then("+").color(ChatColor.AQUA).tooltip(TL.CLAIM_YOUAREHERE.toString());
```

Comment:
The tooltip is contained within an `if (false)` (brb while I git blame that shit, just comment it out ffs.

Reply:
EDIT: https://github.com/drtshock/Factions/commit/fb4292f746bc3de7359bb3acd717fa2f55623b78 IT WAS TRENT!!!!

regardless this should probably be reviewed as to whether it should be turned back on. He probably had a good reason.

Reply:
tooltips caused a bunch of lag, generating and sending huge piles of data, is my understanding

Reply:
Yeeeeeep that makes sense.

So there is a chance to optimize this out by stringing out multiple of the same character using the same tooltip instance - should cut down on that.

Otherwise this needs to be removed (or better yet, configurable!)